File: 3-leetcode/01-easy/js/2053-kth-distinct-string-in-an-array.ts
const kthDistinct = (arr: string[], k: number): string => {
  const map = new Map();

  for (let str of arr) {
    if (!map.has(str)) {
      map.set(str, 1);
      continue;
    }

    map.set(str, map.get(str) + 1);
  }

  for (let str of arr) {
    if (map.get(str) === 1) {
      k--;
    }

    if (k === 0) {
      return str;
    }
  }

  return '';
};

File: 3-leetcode/01-easy/cpp/2053-kth-distinct-string-in-an-array.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution
{
public:
    string kthDistinct(vector<string> &arr, int k)
    {
        unordered_map<string, int> map;

        for (string &str : arr)
        {
            map[str]++;
        }

        for (string &str : arr)
        {
            if (map[str] == 1)
            {
                k--;
            }

            if (k == 0)
            {
                return str;
            }
        }

        return "";
    }
};